tls/s2n_config.h

```diff
@@ -191,6 +194,9 @@ struct s2n_config {
 
     void *renegotiate_request_ctx;
     s2n_renegotiate_request_cb renegotiate_request_cb;
+
+    /* Used for overriding setsockopt calls in testing */
+    s2n_setsockopt_cb setsockopt_cb;
 };
```

Comment:
Because you're putting this on s2n_config instead of s2n_connection (understandable-- memory cost) or making it global, I don't think you can do a blanket override for all tests. What happens if a ktls test forgets to set it?

Reply:
Then it will error with something like "Operation not supported" and the tester can fix the error by configuring it correctly.

Reply:
Are you sure? It won't successfully call the real / "default" setsockopt?

Reply:
The default setsockopt will return the error if you call it accidentally. The reason being you need to have setup the socket correctly before then (connection/accept a connection and be a INET type socket).

```
Error Message: 'An error occurred when attempting to configure the socket for kTLS. Ensure the 'tls' kernel module is enabled.'
 Debug String: 'Error encountered in /home/toidiu/projects/s2n-tls/tls/s2n_ktls.c:239'
 System Error: Operation not supported (95)
```